src/ledger-bridge-keyring.test.ts

```diff
@@ -0,0 +1,866 @@
+            address: fakeAccounts[1],
+          },
+        });
+      });
+
+      await ledgerKeyring.unlock(`m/44'/60'/0'/1`, false);
+      assert.deepEqual(ledgerKeyring.hdk.publicKey, 'ABC');
+    });
+  });
+
+  describe('setHdPath', function () {
+    it('should set the hdPath', function (done) {
+      const someHDPath = `m/44'/99'/0`;
+      keyring.setHdPath(someHDPath);
+      assert.equal(keyring.hdPath, someHDPath);
+      done();
+    });
+
+    it('should reset the HDKey if the path changes', function (done) {
+      const someHDPath = `m/44'/99'/0`;
+      keyring.setHdPath(someHDPath);
+      assert.equal(keyring.hdk.publicKey, null);
+      done();
+    });
+  });
+
+  describe('setAccountToUnlock', function () {
+    it('should set unlockedAccount', function () {
+      keyring.setAccountToUnlock(3);
+      assert.equal(keyring.unlockedAccount, 3);
+    });
+  });
+
+  describe('addAccounts', function () {
+    describe('with no arguments', function () {
+      it('returns a single account', async function () {
+        keyring.setAccountToUnlock(0);
+        const accounts = await keyring.addAccounts();
+        assert.equal(accounts.length, 1);
+      });
+    });
+
+    describe('with a numeric argument', function () {
+      it('returns that number of accounts', async function () {
+        keyring.setAccountToUnlock(0);
+        const accounts = await keyring.addAccounts(5);
+        assert.equal(accounts.length, 5);
+      });
+
+      it('returns the expected accounts', async function () {
+        keyring.setAccountToUnlock(0);
+        const accounts = await keyring.addAccounts(3);
+        assert.equal(accounts[0], fakeAccounts[0]);
+        assert.equal(accounts[1], fakeAccounts[1]);
+        assert.equal(accounts[2], fakeAccounts[2]);
+      });
+    });
+
+    it('stores account details for bip44 accounts', async function () {
+      keyring.setHdPath(`m/44'/60'/0'/0/0`);
+      keyring.setAccountToUnlock(1);
+      sandbox.on(keyring, 'unlock', async () =>
+        Promise.resolve(fakeAccounts[0]),
+      );
+      const accounts = await keyring.addAccounts(1);
+      assert.deepEqual(keyring.accountDetails[accounts[0] as string], {
+        bip44: true,
+        hdPath: `m/44'/60'/1'/0/0`,
+      });
+    });
+
+    it('stores account details for non-bip44 accounts', async function () {
+      keyring.setHdPath(`m/44'/60'/0'`);
+      keyring.setAccountToUnlock(2);
+      const accounts = await keyring.addAccounts(1);
+      assert.deepEqual(keyring.accountDetails[accounts[0] as string], {
+        bip44: false,
+        hdPath: `m/44'/60'/0'/2`,
+      });
+    });
+
+    describe('when called multiple times', function () {
+      it('should not remove existing accounts', async function () {
+        keyring.setAccountToUnlock(0);
+        let accounts = await keyring.addAccounts(1);
+        keyring.setAccountToUnlock(1);
+        accounts = await keyring.addAccounts(1);
```

Comment:
Nit: Maybe better to ignore the first `accounts` returned since they aren't used

```suggestion
        await keyring.addAccounts(1);
        keyring.setAccountToUnlock(1);
        const accounts = await keyring.addAccounts(1);
```